encord/http/querier.py

```diff
@@ -157,10 +151,13 @@ def _do_request(req, timeouts: tuple):
 
         return res_json.get("response")
 
-    @staticmethod
-    def create_new_session() -> Session:
-        session = Session()
 
-        session.mount("https://", HTTPAdapter(max_retries=Retry(connect=0)))
+@contextmanager
+def create_new_session(max_retries: Optional[int] = None, backoff_factor: float = 0) -> Session:
+    retry_policy = Retry(total=None, connect=max_retries, read=max_retries, backoff_factor=backoff_factor)
```

Comment:
to me it is not clear what `status` defaults to so I'd set it to `0` explicitly.

Reply:
Nice one on reading the code. I think if I set the total to max_retries it will limit connect and read combined to max_retries. I'll amend the code.

It will log a debug message.

Updating the docstrings.